### PR TITLE
Bump to `torch` related packages

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

Optuna's PyTorch version looks a little bit old. The latest version is 1.10.0, but the used version is 1.8.0. Indeed, BoTorch drops PyTorch 1.8.* support since the recent version, `0.6.0`, according to its [release note](https://github.com/pytorch/botorch/releases/tag/v0.6.0).

## Description of the changes

- Bump to PyTorch related packages: `torch`, `torchvision`, and `torchaudio`.
- Remove upper constraint of Allennlp. 
- Avoid testing Allennlp with Python 3.6 due to incompatibility; Allennlp 2.8.0 does not support Python 3.6 anymore.

Related issue / context 
- https://github.com/optuna/optuna/issues/3021

### CODE REVIEW

**Comment by nzw0301** [comment]
According to the CI, the stable Alllennlp versions do not support pytorch==1.10.0. I'll update this PR by following the previous bump by https://github.com/optuna/optuna/pull/2442#issuecomment-791055048.

**Comment by nzw0301** [comment]
Ah, allennlp version is also restricted by optuna's `setup.py`.  I'll investigate the possibility to relax the constraint.

File: setup.py [hunk 1]
```diff
@@ -113,12 +113,12 @@ def get_extras_require() -> Dict[str, List[str]]:
...
-            "torchvision==0.9.0+cpu ; sys_platform!='darwin'",
-            "torchaudio==0.8.0",
-            "allennlp>=2.2.0,<2.7.0",
+            "torch==1.10.0 ; sys_platform=='darwin'",
+            "torch==1.10.0+cpu ; sys_platform!='darwin'",
+            "torchvision==0.11.1 ; sys_platform=='darwin'",
+            "torchvision==0.11.1+cpu ; sys_platform!='darwin'",
+            "torchaudio==0.10.0",
+            "allennlp>=2.2.0 ; python_version>'3.6'",
             "botorch>=0.4.0 ; python_version>'3.6'",
```
Comment (himkt) [inline setup.py:122]: [Question] Don't we have to update the lower constraint for botorch as well?

**Review by himkt — APPROVED** [review]
Thank you for the patch. LGTM in the perspective of AllenNLP.

File: setup.py (hunk 1 above)
Comment (nzw0301) [reply setup.py:122]: `pytest tests/integration_tests/test_botorch.py` doesn't show any error with botorch==0.4.0 and botorch==0.5.1.

Comment (himkt) [reply setup.py:122]: Ah, so sorry, you are totally right. I forgot to delete this comment when submitting the review...

Comment (nzw0301) [reply setup.py:122]: No worries. In fact, I did not check them. Fortunately, there was no error :)

**Review by hvy — COMMENTED** [review]
Basically LGTM. I'm just wondering if we must also update the AllenNLP examples in `optuna-examples` to exclude Python 3.6 from the CI matrix https://github.com/optuna/optuna-examples/blob/main/.github/workflows/allennlp.yml. Do you have any insight on this?

**Comment by himkt** [comment]
requirements.txt for AllenNLP example doesn't specify PyTorch so CI would run successfully, but it would be consistent to align depdendencies between `optuna` and `optuna-examples` as you pointed out. @hvy You think we should update optuna-examples, right?

**Comment by hvy** [comment]
Ah, so it's not that the AllenNLP example will start to fail after this change (because of the upper constraint in https://github.com/optuna/optuna-examples/blob/v3.0.0-a0/allennlp/requirements.txt#L1 ?). In any case, yes I think we should keep the aligned, but if that's the case, I think we can go ahead and merge this PR first. Is this understanding correct?

**Comment by himkt** [comment]
>  I think we can go ahead and merge this PR first. Is this understanding correct?

Yes, I think so. :+1:

**Review by hvy — APPROVED** [review]
LGTM!

**Pull request merged**